bridges/evm/contracts/Bridge.sol

```diff
@@ -1,37 +1,57 @@
 pragma solidity 0.5.14;
 pragma experimental ABIEncoderV2;
 import {BlockHeaderMerkleParts} from "./BlockHeaderMerkleParts.sol";
+import {Ownable} from "openzeppelin-solidity/contracts/ownership/Ownable.sol";
 import {IAVLMerklePath} from "./IAVLMerklePath.sol";
 import {TMSignature} from "./TMSignature.sol";
 import {Utils} from "./Utils.sol";
 import {IBridge} from "./IBridge.sol";
 
 /// @title Bridge <3 BandChain D3N
 /// @author Band Protocol Team
-contract Bridge is IBridge {
+contract Bridge is IBridge, Ownable {
     using BlockHeaderMerkleParts for BlockHeaderMerkleParts.Data;
     using IAVLMerklePath for IAVLMerklePath.Data;
     using TMSignature for TMSignature.Data;
 
     /// Mapping from block height to the hash of "zoracle" iAVL Merkle tree.
     mapping(uint256 => bytes32) public oracleStates;
     /// Mapping from an address to whether it's a validator.
-    mapping(address => bool) public validators;
+    mapping(address => uint256) public validatorPowers;
     /// The total number of active validators currently on duty.
-    uint256 public validatorCount;
+    uint256 public totalValidatorPower;
+
+    struct ValidatorWithPower {
+        address addr;
+        uint256 power;
+    }
 
     /// Initializes an oracle bridge to BandChain.
     /// @param _validators The initial set of BandChain active validators.
-    constructor(address[] memory _validators) public {
+    constructor(ValidatorWithPower[] memory _validators) public {
         for (uint256 idx = 0; idx < _validators.length; ++idx) {
-            address validator = _validators[idx];
+            ValidatorWithPower memory validator = _validators[idx];
             require(
-                !validators[validator],
+                validatorPowers[validator.addr] == 0,
                 "DUPLICATION_IN_INITIAL_VALIDATOR_SET"
             );
-            validators[validator] = true;
+            validatorPowers[validator.addr] = validator.power;
+            totalValidatorPower += validator.power;
```

Comment:
Let's use safemath here and everywhere else

Reply:
Yes sir!